### PR TITLE
Profile-qualified model identity in status bar and diagnostics (Fixes #2193)

### LINKED ISSUE #2193: Profile-qualified model identity in status bar and diagnostics
## Summary

The model identity shown in the CLI status/footer should identify the profile context, not just the raw provider/model. This is especially important for load-balancer profiles, where the visible model can otherwise hide which load-balancer profile and which selected sub-profile are actually being used.

## Requested behavior

1. When a normal profile is active, the status bar model display should use:

    profileName:modelName

2. When a load-balancer profile is active, the status bar model display should use:

    lb:loadBalancerProfileName:activeSubProfileName:activeModelName

   The active sub-profile should be the most recently selected load-balancer sub-profile for the current request/session state. The active model name should be the model that will actually be used by that selected sub-profile, not only the load-balancer wrapper's synthetic model name.

3. If no profile is active, keep a clear fallback that preserves the current useful model identity. Do not regress direct provider/model display for non-profile sessions.

4. If a load-balancer profile is active but no sub-profile has been selected yet, show an explicit placeholder such as none or pending rather than silently omitting the load-balancer profile context.

## Diagnostics requirements

The diagnostics output should make load-balancer state unambiguous. When the active provider is a load-balancer profile, diagnostics should include all of the following as distinct, non-duplicated fields:

- Actual active profile name loaded by the CLI.
- Load-balancer profile name as reported by the load-balancer provider stats/config.
- Member sub-profiles included in the load-balancer profile.
- Active/last-selected sub-profile name.
- Active model name for the selected sub-profile.
- Existing request counts/statistics, without duplicating fields under different names.

While implementing this, audit the current diagnostics provider/model/profile fields for duplicated or ambiguous values and make each field unique in meaning. For example, avoid having both Current Profile and Load Balancer Profile print the same value without explaining that one is runtime profile state and the other is provider stats/config state.

## Current code notes from initial analysis

Likely touchpoints:

- packages/cli/src/ui/components/Footer.tsx
  - ModelNameDisplay currently has load-balancer-specific rendering via load-balancer stats.
  - Current load-balancer display appears to render the last selected sub-profile plus a secondary via load-balancer profile label, rather than the requested single identity string.
- packages/cli/src/ui/containers/SessionController.tsx
  - Currently computes the model display as providerName:modelName when both are present.
- packages/cli/src/ui/containers/AppContainer/hooks/useModelRuntimeSync.ts
  - Manages currentModel/currentModelLabel behavior and has relevant tests for profile/provider display behavior.
- packages/cli/src/ui/commands/diagnosticsCommand.ts
  - Prints runtime snapshot provider/model/profile.
  - Has a load-balancer stats section that currently exposes Active Sub-Profile from lastSelected.
- packages/providers/src/LoadBalancingProvider.ts
  - LoadBalancerStats includes profileName, lastSelected, totalRequests, and profileCounts.
  - Extended stats already carry profile/member-related data used for diagnostics and footer display.

## Basic implementation plan

1. Add/adjust behavioral tests for footer/status identity formatting:
   - Standard profile displays profileName:modelName.
   - Direct non-profile session keeps a sensible existing fallback.
   - Load-balancer profile before first selection displays lb:profileName:none-or-pending:modelName-or-pending.
   - Load-balancer profile after selection displays lb:profileName:activeSubProfileName:activeModelName.

2. Add/adjust diagnostics tests:
   - Standard runtime diagnostics still show provider/model/profile clearly.
   - Load-balancer diagnostics show active runtime profile, load-balancer profile name, member sub-profiles, active sub-profile, and active model name.
   - No duplicate/ambiguous diagnostics fields remain in the load-balancer section.

3. Identify a single source of truth for display identity:
   - Prefer runtime/profile state for the loaded profile name.
   - Use load-balancer provider stats/config for load-balancer profile name and last selected sub-profile.
   - Add any missing model/sub-profile metadata to stats or runtime snapshot if necessary instead of reconstructing it in multiple UI locations.

4. Update the footer/status rendering and diagnostics rendering to consume the same normalized identity information where practical, so the status bar and diagnostics agree.

5. Run targeted tests for Footer/SessionController/useModelRuntimeSync and diagnosticsCommand, then the repository verification suite before merging.

## Acceptance criteria

- [ ] Standard active profile status display is profileName:modelName.
- [ ] Load-balancer active profile status display is lb:loadBalancerProfileName:activeSubProfileName:activeModelName.
- [ ] Load-balancer status display has a clear no-selection fallback before the first selected sub-profile.
- [ ] Diagnostics for load-balancer sessions include runtime profile name, load-balancer profile name, member sub-profiles, active sub-profile, and active model name.
- [ ] Diagnostics fields are unique in meaning and do not duplicate the same value under ambiguous labels.
- [ ] Existing non-profile provider/model display remains understandable and covered by tests.
- [ ] Tests cover standard profile, direct provider, load-balancer before selection, and load-balancer after selection.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Fixes #2193.

Makes the model identity shown in the CLI status bar (footer) and `/diagnostics` identify the **profile context**, not just the raw provider/model. This is especially important for load-balancer profiles, where the visible model previously hid which load-balancer profile and which selected sub-profile were actually being used.

## Status bar behavior

A new pure helper `packages/cli/src/ui/utils/modelIdentity.ts` is the single source of truth for the identity string:

- **Standard profile session** → `profileName:modelName` (or just `profileName` when no model is available).
- **Load-balancer session** → `lb:loadBalancerProfile:activeSubProfile:activeModel`.
  - Before the first sub-profile selection, both the sub-profile and model render as an explicit `none` placeholder (`lb:my-lb:none:none`) rather than silently omitting context.
  - The active model is the model the selected sub-profile will actually use, resolved from the load-balancer provider stats — not the synthetic load-balancer wrapper model name.
- **Direct (non-profile) session** → `provider:model` (or `provider` / `model`), with a final `unknown` fallback so non-profile display does not regress.

`formatModelIdentity` builds the string; `resolveModelIdentity` reads the runtime accessors (active provider status, active profile name, and load-balancer stats) and is resilient when the provider manager is unavailable, the load-balancer provider is missing, or `getStats()` returns null/throws — in those cases it still renders the load-balancer context with pending placeholders. The footer's `ModelNameDisplay` now delegates to this helper, replacing the previous ad-hoc `<sub> via <profile>` rendering.

## Load-balancer stats

`ExtendedLoadBalancerStats` now exposes:

- `members: string[]` — the member sub-profile names, in configuration order.
- `lastSelectedModel: string | null` — the model of the most recently selected sub-profile.

Both are derived in `buildExtendedStats` from the resolved sub-profiles (handling both the resolved `model` field and the unresolved `modelId` field). There is **no change** to the selection/increment code paths.

## Diagnostics

The load-balancer section of `/diagnostics` now reports distinct, non-duplicated fields:

- `Load Balancer Profile` (provider stats/config state)
- `Member Sub-Profiles`
- `Active Sub-Profile`
- `Active Model`
- `Total Requests` + `Profile Distribution`

These are clearly disambiguated from the runtime `Current Profile` / `Current Model` lines, addressing the issue's requirement that each field be unique in meaning (e.g. `Current Profile` is runtime profile state while `Load Balancer Profile` is provider stats/config state). `diagnosticsCommand` now consumes the exported `ExtendedLoadBalancerStats` type directly so the output stays coupled to the real provider contract.

`RuntimeContext` additionally exposes `getActiveProfileName` so the UI can read the active runtime profile without reconstructing it.

## Tests (test-first)

- `modelIdentity.test.ts` — standard, direct, model-only, fallback, and load-balancer (before/after selection, empty/whitespace handling) formatting, plus runtime resolution including provider-not-found, getStats-null, and getStats-throws fallbacks.
- `LoadBalancingProvider.activeModel.test.ts` — `members` ordering and `lastSelectedModel` behavior (round-robin selection, resolved sub-profiles, reset).
- `diagnosticsCommand.loadbalancer.spec.ts` — disambiguated, non-duplicated load-balancer diagnostics fields, including the `none` placeholders before first selection.
- Updated the `runtimeSettings.js` mock factory in affected config tests to include the new `getActiveProfileName` export.

## Acceptance criteria

- [x] Standard active profile status display is `profileName:modelName`.
- [x] Load-balancer active profile status display is `lb:loadBalancerProfileName:activeSubProfileName:activeModelName`.
- [x] Load-balancer status display has a clear no-selection fallback before the first selected sub-profile.
- [x] Diagnostics for load-balancer sessions include runtime profile name, load-balancer profile name, member sub-profiles, active sub-profile, and active model name.
- [x] Diagnostics fields are unique in meaning and do not duplicate the same value under ambiguous labels.
- [x] Existing non-profile provider/model display remains understandable and covered by tests.
- [x] Tests cover standard profile, direct provider, load-balancer before selection, and load-balancer after selection.

## Verification

- `npm run build` — pass
- `npm run typecheck` — pass
- `npm run format` — pass
- ESLint on changed files + `npm run lint:eslint-guard` — pass
- Providers full suite — 3475 passed / 18 skipped
- CLI full suite — 4881 passed / 6 skipped
- Smoke test (`ollamakimi` profile) — pass
- Open code review (ocr) — all findings addressed (unsafe casts removed, duplicated stats interfaces replaced with the exported type, additional edge-case tests added).

## Notes

`displayLabel` semantics (used by the chat "Responding with:" notification) are intentionally left unchanged; the footer identity is computed separately via `resolveModelIdentity`, so this change does not regress that path.

### CODE REVIEW

**Comment by acoliver** [comment]
@coderabbitai review

**Comment by acoliver** [comment]
@coderabbitai full review

**Comment by acoliver** [comment]
## CI status — evidence that the two red checks are an external API outage, not this PR

All substantive checks for this PR are green:

- Lint (Javascript / Shell / YAML / GitHub Actions / umbrella Lint): pass
- CodeQL: pass
- Interactive UI (tmux): pass
- Test matrix — Test (ubuntu-latest, keyring), (ubuntu-latest, fallback), (macos-latest, keyring), (macos-latest, fallback): all pass
- CodeRabbit: pass (review completed)

Two checks are red, and both are caused by an upstream LLM API outage rather than anything in this PR:

### 1. E2E Test (macOS / Linux sandbox:none / Linux sandbox:docker)

The E2E suite drives the CLI against the real gemini API. The failing cases are generic, provider-driven tool tests with no relationship to this PR (which only touches load-balancer footer identity + diagnostics):

- list_directory > should be able to list a directory
- replace > should be able to replace content in a file
- session-summary flag > should write a session summary in non-interactive mode

Each fails the same way:

    Error when talking to gemini API ...
    TestRig.run() timed out after 240000ms

i.e. the CLI is waiting 240s per turn for a gemini response that never arrives, which is why the jobs run ~42-46 min (normal is ~8 min) before timing out.

Proof this is environmental and not this PR (verified via gh): the same Testing: E2E workflow is failing with the same ~45 min timeout signature on unrelated PRs at the same time:

- this PR: failure 45m47s
- PR #2203 (Migrate CLI command surfaces to Agent APIs): failure 46m16s
- PR #2199 (Harden agent runtime API boundary): failure 45m36s

The last green Testing: E2E run was the main-merge for #2224 at 8m50s, immediately before the outage began.

### 2. Run LLxprt review

This is the repo's own hosted-LLM PR reviewer. It failed with the same upstream symptom (Error when talking to openai API) and self-reports an infrastructure failure (WARNING: LLxprt PR Review infrastructure failure ... re-run once resolved.). It is also currently failing on the unrelated PR #2228, confirming it is environmental.

### Plan

No code change is warranted; the local verification suite is fully green: typecheck (0 errors), targeted eslint (clean), prettier (clean), build (clean), providers unit/integration (3478 passed), cli unit/integration (4884 passed). I will re-run the failed E2E jobs once the gemini API recovers so the PR can go fully green.

**Comment by acoliver** [comment]
## CI status update — E2E / LLxprt-review failures are a continuing external API outage (not this PR)

All substantive checks on this PR are **green**: CodeQL, all 4 Test-matrix jobs (macOS/Ubuntu × keyring/fallback), all Lint jobs (Javascript/Shell/YAML/GitHub-Actions), Interactive UI (tmux), and CodeRabbit.

The only red checks are **E2E Test (×3)** and **Run LLxprt review**, both of which depend on upstream model APIs that are currently failing platform-wide:

- **E2E**: every E2E run across unrelated PRs is failing with the identical fingerprint:
  - `Error when talking to gemini API`
  - `TestRig.run() timed out after 240000ms`
- **Run LLxprt review**: fails with `Error when talking to openai API` (the reviewer's own hosted-LLM infra, which self-reports an infrastructure failure and asks to re-run once resolved).

### Evidence (gh-verified, cross-PR, continuous ~6h)
The outage reproduces on multiple unrelated PRs, so it is not caused by this branch:

- PR #2199 — E2E failing, same gemini/TestRig fingerprint.
- PR #2203 — E2E failing on a clean re-run (jobs restarted 01:50Z UTC, failed again ~02:05–02:21Z with the same fingerprint).
- PR #2228 — "Run LLxprt review" failing with the same openai error.
- Last green E2E anywhere was pre-outage (~19:42Z UTC). Every E2E run created after ~20:22Z UTC has failed.

### Why this PR's code cannot be the cause
- The only runtime-path change is `LoadBalancingProvider.emitSelectionChanged()` — a synchronous event emit that fires **only** for load-balancer profiles. The E2E suite uses a plain gemini provider, so this code path is never exercised.
- `RetryOrchestrator.getStats()` is an additive delegation outside the generation/retry path.
- The failing E2E tests (generic `list_directory` / `replace` / `session-summary`) are unrelated to model-identity display.

I will re-run the failed E2E and LLxprt-review jobs as soon as the upstream gemini/openai APIs recover (i.e. once any E2E run goes green again).

**Comment by acoliver** [comment]
### CI status update — E2E re-run re-confirms external gemini API outage

I re-ran the failed **E2E** jobs (run 28305507985) at ~03:14Z as a live recovery probe. They again failed (~46 min) with the identical outage fingerprint — this is **not** caused by this PR's changes:

    Error when talking to gemini API
      → TestRig.run() timed out after 240000ms

Fresh evidence from this re-run (E2E Test (Linux) - sandbox:none, job 83872369499):
- 03:29:37Z  Error when talking to gemini API
- 03:32:26Z  TestRig.run() timed out after 240000ms (x3)
- 03:42:15Z  Error when talking to gemini API
- 03:44:48Z  TestRig.run() timed out after 240000ms

Outage scope (unchanged):
- Last green E2E anywhere in the repo: 2026-06-27T19:42:07Z (pre-outage).
- Same failure reproduced on unrelated PRs (#2199, #2203, #2228).
- The repo's own "Run LLxprt review" check fails with the sibling error ("Error when talking to openai API").

Why this PR's code cannot cause it:
- The only runtime change is LoadBalancingProvider.emitSelectionChanged(), a synchronous event emit that fires **only** for load-balancer profiles. E2E uses a plain gemini provider and never exercises this path.
- RetryOrchestrator.getStats() is additive delegation, outside the generation/retry path.
- The failing E2E tests (list_directory / replace / session-summary) are unrelated to model-identity.

All 19 non-gemini-dependent checks pass (CodeQL x2, Test matrix x4, Lint x5, Interactive UI tmux, CodeRabbit, etc.). Will re-run E2E + LLxprt review once the upstream gemini/openai API recovers.

File: packages/providers/src/LoadBalancingProvider.ts [hunk 1]
```diff
@@ -526,9 +529,31 @@ export class LoadBalancingProvider implements IProvider {
...
+   */
+  private emitSelectionChanged(): void {
+    try {
+      coreEvents.emitModelChanged(this.config.profileName);
+    } catch (error) {
+      this.logger.debug(
+        () =>
+          `Failed to emit model-change trigger for load-balancer selection: ${error}`,
+      );
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/LoadBalancingProvider.ts]: Good catch — confirmed and fixed in 9cbe162e4.

You're correct: the selection event only fired from incrementStats() on a successful request. Round-robin calls incrementStats() before delegating (so its footer updated on selection), but the failover strategy only calls incrementStats() after a stream completes successfully. A failing primary was therefore never announced, leaving the footer stale (or "none") for the whole request — exactly the user-reported opusfirst symptom (down primary silently fails over, footer never reflects the backend in use).

Fix splits the two concerns:

- markActiveSelection(): UI-refresh trigger. Sets lastSelected and emits LoadBalancerSelectionChanged when the active backend changes. Idempotent, so retries don't re-emit.
- incrementStats(): success accounting only; delegates selection marking to markActiveSelection().

The failover attempt path now calls markActiveSelection() the moment each backend is chosen, so a failing primary is announced before failover and the footer always reflects the backend actually serving the request. lastSelected now means "most recently selected backend", matching the issue #2193 intent.

Added a test-first failover test (fails the primary, asserts both opusthinking and gpt55high are announced; RED before fix, GREEN after). All 186 LoadBalancingProvider tests pass; core events (17), CLI hook (20), diagnostics LB (4) green.
